test/parallel/test-worker-locks.js

```diff
@@ -0,0 +1,198 @@
+'use strict';
+const { mustCall, mustNotCall } = require('../common');
+const assert = require('assert');
+const { locks, Worker } = require('worker_threads');
+const { EventEmitter, once } = require('events');
+
+const tick = () => new Promise((resolve) => setImmediate(resolve));
```

Comment:
Could just use the awaitable version of `setImmediate()` right?

```suggestion
const { setImmediate: tick } = require('timers/promises')
````

Reply:
Sure, done. :+1: